pkg/connector/client/path.go

```diff
@@ -60,6 +60,9 @@ func withLimitAndOffset(pageToken string, pageSize int) Option {
 func withPaginationCursor(pageSize int,
 	paginationCursor string,
 ) Option {
+	if pageSize < 1 {
+		pageSize = 1
+	}
```

Comment:
We should have a default page size (maybe 100?) and use that if the size is < 1.

Also I think this issue exists because Grants() in `pkg/connector/spaces.go` calls GetSpacePermissions() with pToken.Size, which defaults to zero. The Entitlements() function for spaces does the same thing. I think it's fine to keep those functions the same as long as we have a default page size in the client pkg.

Reply:
Thank you for the inputs! Added the default size, please take another look and let me know if any other changes are required.